File: src/main/java/me/pcasaes/hexoids/core/domain/model/PhysicsQueueEnqueue.java
package me.pcasaes.hexoids.core.domain.model;

import java.util.function.LongPredicate;

public interface PhysicsQueueEnqueue {

    /**
     * Enqueues an action into the physics queue.
     * @param action    Receives timestamp. On return true will re-enqueue to execute on next physics iteration
     */
    void enqueue(LongPredicate action);

}

File: src/main/kotlin/me/pcasaes/hexoids/core/domain/model/PhysicsQueueEnqueue.kt
package me.pcasaes.hexoids.core.domain.model

import java.util.function.LongPredicate

fun interface PhysicsQueueEnqueue {
    /**
     * Enqueues an action into the physics queue.
     * @param action    Receives timestamp. On return true will re-enqueue to execute on next physics iteration
     */
    fun enqueue(action: LongPredicate)
}
